tests/Endpoints/AbstractEndpointTest.php

```diff
@@ -76,4 +76,84 @@ public function testOpaqueIdInHeaders()
         $this->assertNotEmpty($options['client']['headers']['x-opaque-id']);
         $this->assertEquals($params['client']['opaqueId'], $options['client']['headers']['x-opaque-id'][0]);
     }
+
+    /**
+     * @dataProvider deprecationProvider
+     */
+    public function testOldParameterTriggersDeprecation(array $input, bool $shouldThrow, ?string $message): void
```

Comment:
I think it's overcomplicating things. Maybe make it a `testDeprecatedParameterTriggersError` that doesn't take a `shouldThrow`. The happy path with the non-deprecated parameter should just succeed and you expect to get a response from the API when you call, no need to test that it doesn't throw, that's implied.